galsim/config/noise.py

```diff
@@ -506,9 +490,9 @@ def addNoiseVariance(self, config, base, im, include_obj_var, logger):
             # The current image at this point should be the noise-free, sky-free image,
             # which is the object variance in each pixel.
             if gain != 1.0:
-                im += base['current_image']/gain
+                im += base['current_noise_image']/gain
```

Comment:
It likely would be a very minor inefficiency, you're right.  But these little things can add up.  :)